src/components/observer/resize_observer/resize_observer.tsx

```diff
@@ -87,58 +95,90 @@ const makeResizeObserver = (
   return observer;
 };
 
-export const useResizeObserver = (
-  container: Element | null,
-  dimension?: 'width' | 'height'
-) => {
-  const [size, _setSize] = useState({ width: 0, height: 0 });
+export interface UseResizeObserverProps {
+  element?: HTMLElement | null;
+  elementRef?: MutableRefObject<any> | RefObject<any> | null;
+  elementId?: string;
+  observableDimension?: 'all' | 'width' | 'height';
+  shouldObserve?: boolean;
+}
+
+export const useResizeObserver = ({
+  element,
+  elementRef,
+  elementId,
+  observableDimension = 'all',
+  shouldObserve = true,
+}: UseResizeObserverProps) => {
```

Comment:
Changing the prototype like this is a breaking change

Reply:
Yes, I know that this is a breaking change. At the moment I can see two ways around this, as only the interface of this function has changed. The first is to create a new hook, the second is to choose how it is processed depending on the container prop type. Is there another way to around this?